src/Build/BuildCheck/Infrastructure/BuildEventsProcessor.cs

```diff
@@ -49,6 +38,7 @@ public TaskKey(BuildEventContext context)
 
     private readonly SimpleProjectRootElementCache _cache = new SimpleProjectRootElementCache();
     private readonly BuildCheckCentralContext _buildCheckCentralContext = buildCheckCentralContext;
+    private Dictionary<string, (string EnvVarValue, string File, int Line, int Column)> _evaluatedEnvironmentVariables = new Dictionary<string, (string EnvVarValue, string File, int Line, int Column)>();
```

Comment:
Btw. there seems to be 2 levels of deduplication - here in infra and in the check itself - why is that?

Reply:
The different projects may have the same imports and it would cause the duplication of the env variables like these:

```
MSBUILD : warning BC0103: C:\msbuild\msbuild_yk\msbuild\artifacts\bin\bootstrap\net472\MSBuild\Current\Microsoft.Common.props (84,119): https://aka.ms/buildcheck/codes#BC0103 - Property is derived from environment variable: 'MSBuildUserExtensionsPath' with value: 'C:\Users\ykovalova\AppData\Local\Microsoft\MSBuild'. Properties should be passed explicitly using the /p option.
MSBUILD : warning BC0103: C:\msbuild\msbuild_yk\msbuild\artifacts\bin\bootstrap\net472\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets (2578,5): https://aka.ms/buildcheck/codes#BC0103 - Property is derived from environment variable: 'LOCALAPPDATA' with value: 'C:\Users\ykovalova\AppData\Local'. Properties should be passed explicitly using the /p option.
MSBUILD : warning BC0103: C:\Program Files\dotnet\sdk\8.0.400-preview.0.24312.1\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets (97,5): https://aka.ms/buildcheck/codes#BC0103 - Property is derived from environment variable: 'USERPROFILE' with value: 'C:\Users\ykovalova'. Properties should be passed explicitly using the /p option.
```